File: src/main/java/com/senderman/lastkatkabot/command/user/Action.java
package com.senderman.lastkatkabot.command.user;

import com.annimon.tgbotsmodule.api.methods.Methods;
import com.annimon.tgbotsmodule.commands.context.MessageContext;
import com.senderman.lastkatkabot.command.CommandExecutor;
import org.springframework.stereotype.Component;

@Component
public class Action implements CommandExecutor {

    public Action() {
    }

    @Override
    public String getTrigger() {
        return "/action";
    }

    @Override
    public String getDescription() {
        return "сделать действие. Действие указывать через пробел, можно реплаем";
    }

    @Override
    public void execute(MessageContext ctx) {
        ctx.deleteMessage().callAsync(ctx.sender);
        ctx.setArgumentsLimit(1);
        if (ctx.argumentsLength() < 1) return;

        var action = ctx.user().getFirstName() + " " + ctx.argument(0);
        var sm = Methods.sendMessage(ctx.chatId(), action);
        if (ctx.message().isReply()) {
            sm.setReplyToMessageId(ctx.message().getReplyToMessage().getMessageId());
        }
        sm.callAsync(ctx.sender);
    }
}

File: src/main/kotlin/com/senderman/lastkatkabot/usercommands/Action.kt
package com.senderman.lastkatkabot.usercommands

import com.annimon.tgbotsmodule.api.methods.Methods
import com.senderman.lastkatkabot.LastkatkaBotHandler
import com.senderman.neblib.CommandExecutor
import org.telegram.telegrambots.meta.api.objects.Message

class Action(private val handler: LastkatkaBotHandler) : CommandExecutor {

    override val command: String
        get() = "/action"
    override val desc: String
        get() = "сделать действие. Действие указывать через пробел, можно реплаем"

    override fun execute(message: Message) {
        Methods.deleteMessage(message.chatId, message.messageId).call(handler)
        if (message.text.split("\\s+".toRegex()).size == 1) return

        val action = message.text.split("\\s+".toRegex(), 2)[1]
        val sm = Methods.sendMessage(message.chatId, message.from.firstName + " " + action)
        if (message.isReply) sm.replyToMessageId = message.replyToMessage.messageId
        try {
            handler.sendMessage(sm)
        } catch (ignored: Exception) {
        }
    }
}